ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -200,40 +95,48 @@ public KsqlResource(
       final ActivenessRegistrar activenessRegistrar,
       final Function<ServiceContext, SchemaInjector> schemaInjectorFactory
   ) {
-    this.ksqlConfig = Objects.requireNonNull(ksqlConfig, "ksqlConfig");
+    Objects.requireNonNull(ksqlConfig, "ksqlConfig");
+    Objects.requireNonNull(serviceContext, "serviceContext");
+    Objects.requireNonNull(schemaInjectorFactory, "schemaInjectorFactory");
+
     this.ksqlEngine = Objects.requireNonNull(ksqlEngine, "ksqlEngine");
-    this.serviceContext = Objects.requireNonNull(serviceContext, "serviceContext");
     this.commandQueue = Objects.requireNonNull(commandQueue, "commandQueue");
     this.distributedCmdResponseTimeout =
         Objects.requireNonNull(distributedCmdResponseTimeout, "distributedCmdResponseTimeout");
     this.activenessRegistrar =
         Objects.requireNonNull(activenessRegistrar, "activenessRegistrar");
-    this.schemaInjectorFactory =
-        Objects.requireNonNull(schemaInjectorFactory, "schemaInjectorFactory");
-    this.schemaInjector =
-        Objects.requireNonNull(schemaInjectorFactory.apply(serviceContext));
+
+    this.validator = new RequestValidator(
+        CustomValidators.VALIDATOR_MAP,
+        schemaInjectorFactory,
+        ksqlEngine::createSandbox,
+        serviceContext,
```

Comment:
pass a sandboxed service context here - then we _know_ the validator can't change state.